File: ws_test.py
import requests
import websocket
import json
import time

# tạo session mới


SESSION_ID = "478ead8c-e174-43e6-9b1e-c812b207fe77"  # session đã live/start
URL = f"ws://127.0.0.1:8000/ws/state/{SESSION_ID}"

ws = websocket.WebSocket()
ws.connect(URL)

print("✅ Connected to", URL)
count = 0

while True:
    try:
        msg = ws.recv()   # 🔥 DÒNG QUYẾT ĐỊNH
        count += 1

        data = json.loads(msg)

        print(f"\n📩 MESSAGE #{count}")
        print("  type =", data.get("type"))
        print("  mode =", data.get("mode"))
        print("  system.state =", data.get("system", {}).get("state"))
        print("  payload_empty =", data.get("system") == {})
        print(json.dumps(data, indent=2))
    except Exception as e:
        print("❌ WS error:", e)
        break

File: ws_test.js
const WebSocket = require("ws");

const SESSION_ID = "478ead8c-e174-43e6-9b1e-c812b207fe77";
const ws = new WebSocket(`ws://127.0.0.1:8000/ws/state/${SESSION_ID}`);

ws.on("open", () => {
  console.log("✅ WS connected");
});

ws.on("message", (data) => {
  console.log("📩 WS message:", data.toString());
});

ws.on("close", () => {
  console.log("❌ WS closed");
});

ws.on("error", (err) => {
  console.error("🔥 WS error:", err);
});
